gtk/src/callbacks.c

```diff
@@ -577,6 +579,70 @@ check_name_template(signal_user_data_t *ud, const char *str)
     return FALSE;
 }
 
+int match(const char *string, const char *pattern)
+{
+    int status;
+    regex_t re;
+    if (regcomp(&re, pattern, REG_EXTENDED|REG_NOSUB) != 0)
+    {
+        return(0);
+    }
+    status = regexec(&re, string, (size_t) 0, NULL, 0);
+    regfree(&re);
+    if (status != 0) 
+    {
+        return(0);
+    }
+    return(1);
+}
+
+typedef struct {
+    const char *pattern;
+    const char *format;
+} datemap;
+
+static int
+parse_datestring(const char *src, struct tm *tm)
+{       
+    datemap ymdThmsZ = {"[0-9]{4}-[0-1]?[0-9]-[0-3]?[0-9]T[0-9]{2}:[0-9]{2}:[0-9]{2}Z", "%Y-%m-%dT%H:%M:%SZ"};  
+
+    datemap maps[1] = { ymdThmsZ };
+
+    for (int i = 0; i < sizeof(maps); i++)
+    {
+        if (match(src, maps[i].pattern))
+        {
+            strptime(src, maps[i].format, tm);
```

Comment:
warning: implicit declaration of function ‘strptime’

strptime requires _XOPEN_SOURCE define somewhere.  This definition really belongs in gtk/config.h which is generated by autoconf.  So I can make the necessary modification to configure.ac if you don't know autoconf.

Reply:
@jstebbins I think I will need some help on this one. I made the fixes requested here, and also pulled and merged the latest changes from upstream and can't get anything to build now. TBH I had to create a new VM so there's also a good chance I've got other dependency issues going on. I keep having issues now with the automake recipe failing and wondering if this might be related?